collect_app/src/test/java/org/odk/collect/android/widgets/GeoPointWidgetTest.java

```diff
@@ -1,85 +1,301 @@
+        GeoPointWidget widget = createWidget(prompt);
+        stubLocationPermissions(widget, true);
+        widget.startGeoButton.performClick();
+
+        verify(waitingForDataRegistry).waitForData(prompt.getIndex());
+    }
+
+    @Test
+    public void whenPromptDoesNotHaveAnswer_buttonShouldLaunchCorrectIntent() {
+        GeoPointWidget widget = createWidget(promptWithAnswer(null));
+        stubLocationPermissions(widget, true);
+        widget.startGeoButton.performClick();
+
+        Intent startedIntent = shadowOf(widgetTestActivity()).getNextStartedActivity();
+        Bundle bundle = startedIntent.getExtras();
+
+        assertThat(startedIntent.getComponent(), equalTo(new ComponentName(widgetTestActivity(), GeoPointActivity.class)));
+        assertBundleArgumentEquals(bundle, null, DEFAULT_LOCATION_ACCURACY, false, true);
+    }
+
+    @Test
+    public void whenPromptHasAnswer_buttonShouldLaunchCorrectIntent() {
+        GeoPointWidget widget = createWidget(promptWithAnswer(new StringData(answer)));
+        stubLocationPermissions(widget, true);
+        widget.startGeoButton.performClick();
+
+        Intent startedIntent = shadowOf(widgetTestActivity()).getNextStartedActivity();
+        Bundle bundle = startedIntent.getExtras();
+
+        assertThat(startedIntent.getComponent(), equalTo(new ComponentName(widgetTestActivity(), GeoPointActivity.class)));
+        assertBundleArgumentEquals(bundle, GeoWidgetUtils.getLocationParamsFromStringAnswer(answer), DEFAULT_LOCATION_ACCURACY, false, true);
+    }
+
+    @Test
+    public void ifWidgetHasPlacementMaps_buttonShouldLaunchCorrectIntent() {
+        when(mapConfigurator.isAvailable(any())).thenReturn(true);
+        GeoPointWidget widget = createWidget(promptWithAppearance(PLACEMENT_MAP, false));
+        stubLocationPermissions(widget, true);
+        widget.startGeoButton.performClick();
+
+        Intent startedIntent = shadowOf(widgetTestActivity()).getNextStartedActivity();
+        Bundle bundle = startedIntent.getExtras();
+
+        assertThat(startedIntent.getComponent(), equalTo(new ComponentName(widgetTestActivity(), GeoPointMapActivity.class)));
+        assertBundleArgumentEquals(bundle, null, DEFAULT_LOCATION_ACCURACY, false, true);
+    }
+
+    @Test
+    public void ifWidgetHasMaps_buttonShouldLaunchCorrectIntent() {
+        when(mapConfigurator.isAvailable(any())).thenReturn(true);
+        GeoPointWidget widget = createWidget(promptWithAppearance(MAPS, false));
+        stubLocationPermissions(widget, true);
+        widget.startGeoButton.performClick();
+
+        Intent startedIntent = shadowOf(widgetTestActivity()).getNextStartedActivity();
+        Bundle bundle = startedIntent.getExtras();
+
+        assertThat(startedIntent.getComponent(), equalTo(new ComponentName(widgetTestActivity(), GeoPointMapActivity.class)));
+        assertBundleArgumentEquals(bundle, null, DEFAULT_LOCATION_ACCURACY, false, false);
+    }
+
```

Comment:
I think there's a test missing here for if there's a map and it's read-only.

Reply:
Sorry @lognaturel I'm not understanding how I would confirm here if the map is read only.